python/indy-agent.py

```diff
@@ -63,32 +62,36 @@
     """ Message processing loop task.
         Message routes are also defined here through the message router.
     """
-
-
     msg_router = agent['msg_router']
     msg_receiver = agent['msg_receiver']
     ui_event_queue = agent['ui_event_queue']
 
     await msg_router.register(CONN.REQUEST, connection.handle_request)
     await msg_router.register(CONN.RESPONSE, connection.handle_response)
+    await msg_router.register(CONN.OFFER, connection.offer_recv)
+    await msg_router.register(CONN.ACKNOWLEDGE, connection.offer_accepted)
+    await msg_router.register(CONN.SENDER_REJECTION, connection.receiver_offer_rejected)
+    await msg_router.register(CONN.RECEIVER_REJECTION, connection.sender_offer_rejected)
+    await msg_router.register(CONN.REJECTION, connection.conn_rejected)
 
     while True:
         encrypted_msg_bytes = await msg_receiver.recv()
-
+        # TODO: make this logic work
```

Comment:
This  section is also closely tied to evolving understanding of how agents should receive and process messages. First of all, all communications received by an agent are Anonymously encrypted using a public key that is communicated in the connection offer and then sent to the endpoint also specified in that connection offer. That's why the decryption was happening here.

Our UI doesn't know how to anonymously encrypt information for an endpoint and therefore, the UI should never POST to another agent's `/indy` enpoint directly. Instead, a message is sent to the "back-end", which does know how to encrypt using the libindy `anon_encrypt`, and instructs it to send the encrypted message to the other agent's endpoint.

Combining the `/indy` and `/offer` endpoints may be the cause of the difficulties here.